File: frontend/src/types/activity.ts
export interface ActivityLogEntry {
  id: string;
  board_id: string;
  card_id: string | null;
  actor_id: string | null;
  automation_rule_id: string | null;
  action_type: string;
  payload: Record<string, unknown>;
  created_at: string;
}

File: backend/app/schemas/activity.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ActivityLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    board_id: uuid.UUID
    card_id: uuid.UUID | None
    actor_id: uuid.UUID | None
    automation_rule_id: uuid.UUID | None
    action_type: str
    payload: dict
    created_at: datetime
